File: model/Pocket.js
class Pocket {
    constructor(number) {
        this._objects = [];
        this._number = number;
    }

    getObjects() {
        return this._objects;
    }

    getNumber() {
        return this._number;
    }

    empty() {
        this._objects = [];
    }

    putObject(obj) {
        this._objects.push(obj);
    }

    retrieveObject(obj) {
        const index = this._objects.indexOf(obj);
        if (index !== -1) {
            this._objects.splice(index, 1);
        }
    }

    toString() {
        return `Pocket{objects=${this._objects}, number=${this._number}}`;
    }
}


export default Pocket;

File: model/Pocket.cs
using System;
using System.Collections.Generic;


public class Pocket
{
    private readonly List<PocketObject> objects;
    private readonly int number;

    public Pocket(int number)
    {
        this.objects = new List<PocketObject>();
        this.number = number;
    }

    public List<PocketObject> GetObjects()
    {
        return objects;
    }

    public int GetNumber()
    {
        return number;
    }

    public void Empty()
    {
        objects.Clear();
    }

    public void PutObject(PocketObject obj)
    {
        objects.Add(obj);
    }

    public void RetrieveObject(PocketObject obj)
    {
        objects.Remove(obj);
    }
}
